src/Compression/Compression.cs

```diff
@@ -0,0 +1,758 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.IO;
+using System.IO.Compression;
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Compression
+{
+	public partial class CompressionStream : Stream
+	{
+		private const int DefaultBufferSize = 8192;
+
+		private Stream _stream;
+		private CompressionMode _mode;
+		private bool _leaveOpen;
+		private Inflater _inflater;
+		private Deflater _deflater;
+		private byte[] _buffer;
+		private int _activeAsyncOperation; // 1 == true, 0 == false
+		private bool _wroteBytes;
+
+		public CompressionStream (Stream stream, CompressionMode mode) : this(stream, mode, algorithm: CompressionAlgorithm.Zlib, leaveOpen: false)
+		{
+		}
```

Comment:
I"m not sure overloads without a `CompressionAlgorithm` are very useful since they require going to documentation (not existing yet) to know which algorithm is the default. Simpler (and faster) to have it provided.

Reply:
The default out is zlib, this was done to be able to use the DeflateStream tests without changes. I can remove the constructors without the algo, but will required an update in the tests.